sdk/storage/azure-storage-file-datalake/tests/test_datalake_service_client.py

```diff
@@ -452,3 +452,50 @@ def test_datalake_clients_properly_close(self, **kwargs):
         dir_client._datalake_client_for_blob_operation.close.assert_called_once()
         file_client._client.__exit__.assert_called_once()
         file_client._datalake_client_for_blob_operation.close.assert_called_once()
+
+    @DataLakePreparer()
+    @recorded_by_proxy
+    def test_storage_account_audience_service_client(self, **kwargs):
+        datalake_storage_account_name = kwargs.pop("datalake_storage_account_name")
+        datalake_storage_account_key = kwargs.pop("datalake_storage_account_key")
+
+        # Arrange
+        self._setup(datalake_storage_account_name, datalake_storage_account_key)
+        self.dsc.create_file_system('testfs1')
+
+        # Act
+        token_credential = self.generate_oauth_token()
+        dsc = DataLakeServiceClient(
+            self.account_url(datalake_storage_account_name, "blob"),
+            credential=token_credential,
+            audience=f'https://{datalake_storage_account_name}.blob.core.windows.net/'
```

Comment:
Datalake tests should use datalake endpoint as the audience.

Also, does this test test one operation that uses the Blob endpoint and one that uses dfs? Thinking about it, we probably need to convert a dfs audience to a blob one for the Blob client constructor?

Reply:
Yep! 
List -> Blob
Create -> Datalake